train.py

```diff
@@ -67,35 +78,46 @@
-                print('saving the latest model (epoch %d, total_iters %d)' % (epoch, total_iters))
-                save_suffix = 'iter_%d' % total_iters if opt.save_by_iter else 'latest'
-                model.save_networks(save_suffix)
-
-            if total_iters % opt.fid_every < model.opt.batch_size and opt.compute_fid:
-                model.compute_fid(epoch,total_iters)
-                if opt.display_id > 0:
-                    fids=model.get_current_fids()
-                    visualizer.plot_current_fid(epoch, float(epoch_iter) / dataset_size, fids)
+
+            if rank == 0:
+                if total_iters % opt.display_freq < model.opt.batch_size:   # display images on visdom and save images to a HTML file
+                    save_result = total_iters % opt.update_html_freq == 0
+                    model.compute_visuals()
+                    visualizer.display_current_results(model.get_current_visuals(), epoch, save_result,params=model.get_display_param())
+
+                if total_iters % opt.print_freq < model.opt.batch_size :    # print training losses and save logging information to the disk
+                    losses = model.get_current_losses()
+                    visualizer.print_current_losses(epoch, epoch_iter, losses, t_comp, t_data_mini_batch)
+                    if opt.display_id > 0:
+                        visualizer.plot_current_losses(epoch, float(epoch_iter) / dataset_size, losses)
+
+                if total_iters % opt.save_latest_freq < model.opt.batch_size :   # cache our latest model every <save_latest_freq> iterations
+                    print('saving the latest model (epoch %d, total_iters %d)' % (epoch, total_iters))
+                    save_suffix = 'iter_%d' % total_iters if opt.save_by_iter else 'latest'
+                    model.save_networks(save_suffix)
+
+                if total_iters % opt.fid_every < model.opt.batch_size and opt.compute_fid:
+                    model.compute_fid(epoch,total_iters)
+                    if opt.display_id > 0:
+                        fids=model.get_current_fids()
+                        visualizer.plot_current_fid(epoch, float(epoch_iter) / dataset_size, fids)
     
-            iter_data_time = time.time()
+                iter_data_time = time.time()
             
-        if epoch % opt.save_epoch_freq == 0:              # cache our model every <save_epoch_freq> epochs
-            print('saving the model at the end of epoch %d, iters %d' % (epoch, total_iters))
-            model.save_networks('latest')
-            model.save_networks(epoch)
+            if epoch % opt.save_epoch_freq == 0:              # cache our model every <save_epoch_freq> epochs
+                print('saving the model at the end of epoch %d, iters %d' % (epoch, total_iters))
+                model.save_networks('latest')
+                model.save_networks(epoch)
```

Comment:
I think only model on rank 0 should be saved? (since they are all the same)
It looks like this should be in the `if rank == 0` conditional